File: JavaScript/DS/queue.js
/**
 * Class representing a Queue data structure.
 */
class Queue {
    /**
     * Create an empty queue.
     */
    constructor() {
      this.items = [];
    }
  
    /**
     * Enqueue an item into the queue.
     * @param {any} item - The item to be enqueued.
     */
    enqueue(item) {
      this.items.push(item);
    }
  
    /**
     * Dequeue and return the front item from the queue.
     * @returns {any} - The front item from the queue.
     */
    dequeue() {
      if (!this.isEmpty()) {
        return this.items.shift();
      } else {
        return null;
      }
    }
  
    /**
     * Check if the queue is empty.
     * @returns {boolean} - True if the queue is empty, otherwise false.
     */
    isEmpty() {
      return this.items.length === 0;
    }
  
    /**
     * Return the size (number of items) of the queue.
     * @returns {number} - The size of the queue.
     */
    size() {
      return this.items.length;
    }
  }
  
  module.exports = Queue;
  
  // Example usage:
  if (require.main === module) {
    const queue = new Queue();
    queue.enqueue(1);
    queue.enqueue(2);
    queue.enqueue(3);
    console.log("Queue:", queue.items);
    console.log("Dequeue:", queue.dequeue());
    console.log("Queue Size:", queue.size());
  }
  

File: Python/DS/queue.py
from collections import deque

class Queue:
    """
    Queue class represents a queue data structure.
    It uses a deque from the collections module.
    It contains methods for enqueueing, dequeueing,
    checking if it's empty, and getting the size of the queue.
    """
    def __init__(self):
        self.items = deque()

    def enqueue(self, item):
        """
        Enqueue an item into the queue.
        """
        self.items.append(item)

    def dequeue(self):
        """
        Dequeue and return the front item from the queue.
        """
        if not self.is_empty():
            return self.items.popleft()
        else:
            return None

    def is_empty(self):
        """
        Check if the queue is empty.
        """
        return len(self.items) == 0

    def size(self):
        """
        Return the size (number of items) of the queue.
        """
        return len(self.items)

# Example usage:
if __name__ == "__main__":
    queue = Queue()
    queue.enqueue(1)
    queue.enqueue(2)
    queue.enqueue(3)
    print("Queue:", list(queue.items))
    print("Dequeue:", queue.dequeue())
    print("Queue Size:", queue.size())
